### PR TITLE
Additional test cases added to cover corner cases validation

### PR DESCRIPTION
**Pull request by william-Johnason**
Here is the list of changes:

-  test_wiki.py -> unicode/CJK round-trip, overwrite replaces content, empty content, categories round-trip, missing categories defaults to [], path traversal rejection, orphan flag, set_page_categories replaces (not appends), append_to_index dedup, append_to_index creates ## Recently Added, concurrent writes to different pages.

-  test_query_agent.py: empty wiki returns gracefully, invalid JSON terms falls back to word split, citations match BM25 results, tokens_used is sum of both LLM calls, question string preserved, multi-page citations all exist in store.

-  test_queue.py: exact max-retries → dead, below max-retries → pending, skip prevents retry, fail_permanent → failed not dead, empty dequeue returns None, FIFO order, enqueue_many all pending, retry resets counter, purge removes old, purge keeps recent.
-  update README.md to show All test cases passed/failed.
-  update .gitignore to exclude *.docx file extension

### CODE REVIEW

File: scripts/update_badges.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# Copyright (C) 2026 Paul Chen / axoviq.com
```
Comment (paulmchen) [inline scripts/update_badges.py]: Use your name instead.

File: README.md [hunk 2]
```diff
@@ -19,6 +19,14 @@
   Domain-agnostic LLM wiki engine
 ```
```
Comment (paulmchen) [inline README.md:21]: should we hard code version number in the banner, or it should get it from somewhere?

File: tests/agents/test_query_agent.py [hunk 3]
```diff
@@ -9,6 +9,18 @@
 from synthadoc.core.cache import CacheManager
```
Comment (paulmchen) [inline tests/agents/test_query_agent.py:10]: we don't have enough test cases cover Obsidiant plugins. We should make a note and discuss later how we improve.

**Review by paulmchen — CHANGES_REQUESTED** [review]

**Comment by william-Johnason** [comment]
The comments have been addressed. We do want to add more tests to the plugins. By the way, we can add a few more commands to the plugins. Audit, schedule, scaffod, etc.

File: tests/agents/test_query_agent.py (hunk 3 above)
Comment (william-Johnason) [reply tests/agents/test_query_agent.py:10]: right, we can add a few more commands to the plugins. Audit, schedule, scaffod, etc.

Comment (paulmchen) [reply tests/agents/test_query_agent.py:10]: ok

File: README.md (hunk 2 above)
Comment (william-Johnason) [reply README.md:21]: Version numbers should mostly be changed manually. However, I have fixed tab issues in the readme, so now the contents can be accessed by clicking on the 'REAME', "Contributing" and "Code of Conduct" tabs.

Comment (paulmchen) [reply README.md:21]: It makes sense.

**Review by paulmchen — APPROVED** [review]
approved.
